### PR TITLE
added Fix for VRM Portal and update-proof monitoring & installation scripts

### PR DESCRIPTION
**Pull request by christian1980nrw**
see issue

https://github.com/mitchese/shm-et340/issues/3

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+mkdir /data/drivers/shm-et340
```
Comment (ricott) [inline install.sh:3]: Fails if the `drivers` folder is missing, which was my case

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+chmod +x ./shm-et340
+cd /data/drivers/shm-et340/service
+wget https://raw.githubusercontent.com/christian1980nrw/shm-et340-fixed-vrm-portal/master/service/run
+chmod +x ./run
+ln -s  /data/drivers/shm-et340/service /service/shm-et340
+echo >> /data/rc.local
+echo "ln -s /data/drivers/shm-et340/service /service/shm-et340" >> /data/rc.local
+chmod +x /data/rc.local
+echo Installation finished. Please edit the file /data/drivers/shm-et340/service/run and change the IP adress of your smartmeter.
+echo Please ensure (in your routers DHCP-service) that it is always getting the same IP adress.
```
Comment (ricott) [inline install.sh:16]: Fails with 
```
syntax error near unexpected token `('
```

**Comment by ricott** [comment]
Seem to solve the problem with vrm dashboard not updating 👍
The `run` script should maybe fail if the IP address is not added, ie `ip_of_shm2_0` should be blank to start with and a condition in the loop? 
I ran the `install.sh` script a few times and forgot to change the IP address the last time. This results in the app being restarted each 15-25 seconds. At least I think it was restarted since I could observe the remote console grid meter view "flash" with that interval. With the correct IP address, it works as expected.

**Comment by christian1980nrw** [comment]
> Seem to solve the problem with vrm dashboard not updating 👍 The `run` script should maybe fail if the IP address is not added, ie `ip_of_shm2_0` should be blank to start with and a condition in the loop? I ran the `install.sh` script a few times and forgot to change the IP address the last time. This results in the app being restarted each 15-25 seconds. At least I think it was restarted since I could observe the remote console grid meter view "flash" with that interval. With the correct IP address, it works as expected.

Yes, i know. If the IP is offline or wrong, the service will be restarted every few seconds. Its not a bug, its a feature.
But by default, i can set it to 0.0.0.0.0 and if not corrected the whole service will not be started.

**Comment by christian1980nrw** [comment]
> Sad that there is no way to contact you. Im very new, but I managed to do all your steps.. my problem is, that I run the commands on my PC at root. If I close the concole Cerbo stops the work and is not doing it alone. I Know im dumb, but Maybe you can help me get this thing running alone

@Oldhyper Sadly mitcheese doesnt seem to be active anymore. Try to execute the commands of my install script at your shell. If you execute the commands, the driver should start automatically after a restart of your cerbo. 
https://github.com/christian1980nrw/shm-et340-fixed-vrm-portal/blob/master/install.sh

**Comment by mitchese** [comment]
a bit of a hiatus, but the new dynamic ess has awoken my interest again in this :)

**Comment by christian1980nrw** [comment]
> a bit of a hiatus, but the new dynamic ess has awoken my interest again in this :)

I migrated to another and cheaper meter some time ago and sold the shm 2.0 but your driver was always working well.  This is my new one and working perfect too:  https://github.com/christian1980nrw/VenusOS-SDM72D-M-V2-Eastron-Modbus-driver

If you merged my scripts there is no need to keep the old fork. Thanks for your work.  

Regards Christian

**Comment by christian1980nrw** [comment]
@mitchese please have a look at the install.sh . You have to correct the link to my fork within it https://raw.githubusercontent.com/christian1980nrw/shm-et340-fixed-vrm-portal/master/service/run

**Comment by christian1980nrw** [comment]
> neat, I've also come across [this](https://www.youtube.com/watch?v=gXUjioDFYBg) as another alternative for getting random meters into vrm. I'll correct the link, thanks for pointing that out :)
> 
> and thanks for the contribution!


@mitchese 
...my infrared port is already used by another device (Tibber Pulse). My energy company is getting realtime data with this device, so the 50 EUR 3 phase SDM72D was the cheapest way and is stable since many months.   

If you have time, please also have a look at my repo "Spotmarket-Switcher " .

Regards
Christian

**Comment by mitchese** [comment]
nice! I also have Tibber (Grüße aus Köln), but we're using it with Home Assistant. I'll take a look through your code and use it as inspiration - we have a similar setup using home assistant to trigger loads like our EVSE on/off. I use modbus tcp to write registers in victron to enable/disable charging, command it to buy from the grid, or hold grid meter at 0, etc.... but my setup doesn't quite work how I want and often does the wrong thing and I'm manually toggling things on and off... need some good inspiration!

**Comment by christian1980nrw** [comment]
I was planning to integrate Modus TCP too but unfortunately i dont have time at the moment, so development is paused... Latest version is at my dev branch. Currently its running directly on the cerbo or raspberry.  

 Grüße aus Arnsberg!

**Pull request merged**